docs/book/src/migration/v2vsv3.md

```diff
@@ -2,21 +2,90 @@
+    
+- Upgrade of the version `gcr.io/kubebuilder/kube-rbac-proxy`, which is an optional component enabled by default to secure the request made against the manager, from `0.5.0` to `0.8.0` to address security concerns. The details of all changes can be found in [kube-rbac-proxy][kube-rbac-proxy].
+ 
+## TL;DR of the New `go/v3` Plugin
+
+***More details on this can be found at [here][kb-releases], but for the highlights, check below***
+
+<aside class="note">
+<h1>Default plugin</h1>
+Projects scaffolded with Kubebuilder v3 will use the `go.kubebuilder.io/v3` plugin by default.
+</aside>
+
+- Scaffolded/Generated API version changes:
+  * Use `apiextensions/v1` for generated CRDs (`apiextensions/v1beta1` was deprecated in Kubernetes `1.16`)
+  * Use `admissionregistration.k8s.io/v1` for generated webhooks (`admissionregistration.k8s.io/v1beta1` was deprecated in Kubernetes `1.16`)
+  * Use `cert-manager.io/v1` for the certificate manager when webhooks are used (`cert-manager.io/v1alpha2` was deprecated in `Cert-Manager 0.14`. More info: [CertManager v1.0 docs][cert-manager-docs])
+  
+- Code changes:
+  * The manager flags `--metrics-addr` and `enable-leader-election` now are named `--metrics-bind-address` and `--leader-elect` to be more aligned with core Kubernetes Components. More info: [#1839][issue-1893] 
+  * Liveness and Readiness probes are now added by default using [`healthz.Ping`][healthz-ping].
+  * A new option to create the projects using ComponentConfig is introduced. For more info see its [enhancement proposal][enhancement proposal] and the [Component config tutorial][component-config-tutorial]
+  * Manager manifests now use `SecurityContext` to address security concerns. More info: [#1637][issue-1637] 
+- Misc:
+  * Support for [controller-tools][controller-tools] `v0.4.1` (for `go/v2` it is `v0.3.0` and previously it was `v0.2.5`) 
+  * Support for [controller-runtime][controller-runtime] `v0.7.0` (for `go/v2` it is `v0.6.4` and previously it was `v0.5.0`)
+  * Support for [kustomize][kustomize] `v3.8.7` (for `go/v2` it is `v3.5.4` and previously it was `v3.1.0`)
+  * Required Envtest binaries are automatically downloaded
+  * The minimum Go version is now `1.15` (previously it was `1.13).
+
+<aside class="note warning">
+<h1>Project customizations</h1>
+
+Note that after you use the CLI to create your projects and generate the manifests, you can customise it as please you. Furthermore, would be recommended to not deviate from the layout proposed unless you know what you are doing. 
+
+See that if you switch the places of what is scaffolded by default, it might get yourself in a more challenging position in the future to keep your project maintainable. And then, you might be no longer able to follow up a straightforward process to solve tech-debts and upgrade your projects.
```

Comment:
these two lines seem really messy. Maybe `Deviating too much from the default project layout may cause issues with further use of the SDK to upgrade or manage your project. While these options may still be possible, it might now be a simple or straightforward process. Therefore, it is recommended you only do this if you really know what you're doing. For further information on the project layout, see the doc "[What's in a basic project?][basic-project-doc]".`

Reply:
Why do we even mention changing layout at all?

Reply:
It is very important because we are asking for users re-create their project and they have a unique chance to fix their customizations on top of it in order to keep their project maintainable and upgradable easily by the tool features and helpers. 

In the future, we want to provide a command/tooling to do that as well. Also, see that I clarify indeed this motivation in the header of the manual guide.  


"Usually you only will try to do it manually if you customized your project and deviated too much from the proposed scaffold then, before continue, to ensure that you understand the note about project customizations in the [differences between Kubebuilder v2 and v3][migration-v2vsv3]. Note that you might need to spend more effort to do this process manually than organize your project customizations to follow up the proposed layout and keep your project maintainable and upgradable with less effort in the future. "

Reply:
How about this:
> After using the CLI to create your project, you are free to customise how you see fit. Bear in mind, that it is not recommended to deviate from the proposed layout unless you know what you are doing.
>
>For example, you should refrain from moving the scaffolded files, doing so will make it difficult in upgrading your project in the future. You may also lose the ability to use some of the CLI features and helpers. For further information on the project layout, see the doc "[What's in a basic project?][basic-project-doc]

Reply:
done. tks for all folks